test/executable/executableTests.ts

```diff
@@ -233,230 +215,180 @@ describe("Executable", function(this: Mocha.ISuiteCallbackContext) {
-                assert.strictEqual(err.code, 1, "error code should be 1");
-                done();
-            });
+        it("exits with error if passed directory does not exist", async () => {
+            const status = await execRunner(
+                {config: "test/files/tsconfig-test/tslint.json", project: "test/files/non-existent"},
+            );
+            assert.equal(status, Status.FatalError, "exit code should be 1");
         });
 
-        it("exits with code 1 if file is not included in project", (done) => {
-            execCli(
-                [
-                    "-c",
-                    "test/files/tsconfig-test/tslint.json",
-                    "--project",
-                    "test/files/tsconfig-test/tsconfig.json",
-                    "test/files/tsconfig-test/other.test.ts",
-                ],
-                (err) => {
-                    assert.isNotNull(err, "process should exit with error");
-                    assert.strictEqual(err.code, 1, "error code should be 1");
-                    done();
-                });
+        it("exits with code 1 if file is not included in project", async () => {
+            const status = await execRunner(
+                {
+                    config: "test/files/tsconfig-test/tslint.json",
+                    files: ["test/files/tsconfig-test/other.test.ts"],
+                    project: "test/files/tsconfig-test/tsconfig.json",
+                },
+            );
+            assert.equal(status, Status.FatalError, "exit code should be 1");
         });
 
-        it("exits with code 0 if `tsconfig.json` is passed but it includes no ts files", (done) => {
-            execCli(
-                ["-c", "test/files/tsconfig-no-ts-files/tslint.json", "-p", "test/files/tsconfig-no-ts-files/tsconfig.json"],
-                (err) => {
-                    assert.isNull(err, "process should exit without an error");
-                    done();
-                });
+        it("exits with code 0 if `tsconfig.json` is passed but it includes no ts files", async () => {
+            const status = await execRunner(
+                {config: "test/files/tsconfig-no-ts-files/tslint.json", project: "test/files/tsconfig-no-ts-files/tsconfig.json"},
+            );
+            assert.equal(status, Status.Ok, "process should exit without an error");
         });
 
-        it("can extend `tsconfig.json` with relative path", (done) => {
-            execCli(
-                ["-c", "test/files/tsconfig-extends-relative/tslint-ok.json", "-p",
-                 "test/files/tsconfig-extends-relative/test/tsconfig.json"],
-                (err) => {
-                    assert.isNull(err, "process should exit without an error");
-                    done();
-                });
-        });
+        it("can extend `tsconfig.json` with relative path", async () => {
+            let status = await execRunner(
```

Comment:
minor: prefer `const status1`, `const status2` variables instead of reassigning this one